.github/workflows/golang.yaml

```diff
@@ -68,6 +68,13 @@ jobs:
     steps:
     - uses: actions/checkout@v4
       name: Checkout code
-
-    - name: Build
-      run: make docker-build
+    - name: Get Golang version
+      id: vars
+      run: |
+        GOLANG_VERSION=$( grep "GOLANG_VERSION ?=" versions.mk )
+        echo "GOLANG_VERSION=${GOLANG_VERSION##GOLANG_VERSION ?= }" >> $GITHUB_ENV
+    - name: Install Go
+      uses: actions/setup-go@v5
+      with:
+        go-version: ${{ env.GOLANG_VERSION }}
+    - run: make build
```

Comment:
Is any of this necessary if the default image is still the k8s-infra image? How do we trigger this in cases when the k8s-infra image isnÄt good enough?

Reply:
I prefer it like this, there is nothing special needed to build/compile this project, so defaulting to the upstream Go action is better and makes things easier to manage.

Reply:
Meaning installing go and running `make build` instead of `make docker-build`? I don't have a preference either way.

Reply:
yes, the upstream GitHub Action installs the proper GO version and we can run `make build`